GitUI/UserControls/RevisionGrid.cs

```diff
@@ -1913,25 +1905,23 @@ private void RevisionsCellFormatting(object sender, DataGridViewCellFormattingEv
         /// <param name="totalRowCount">check if grid has changed while thread is queued</param>
         /// <param name="colIndex"></param>
         /// <param name="rowIndex"></param>
-        private void LoadIsMultilineMessageInfo(GitRevision revision, int colIndex, int rowIndex, int totalRowCount, GitModule aModule)
+        private async Task LoadIsMultilineMessageInfoAsync(GitRevision revision, int colIndex, int rowIndex, int totalRowCount)
         {
-            // code taken from CommitInfo.cs
-            CommitData commitData = _commitDataManager.CreateFromRevision(revision);
-            string error = "";
+            var commitData = _commitDataManager.CreateFromRevision(revision);
+
             if (revision.Body == null)
             {
-                _commitDataManager.UpdateCommitMessage(commitData, revision.Guid, ref error);
+                // TODO any error message response is ignored here
+                await _commitDataManager.UpdateCommitMessageAsync(commitData, revision.Guid);
+
                 revision.Body = commitData.Body;
             }
 
             // now that Body is filled (not null anymore) the revision grid can be refreshed to display the new information
             this.InvokeAsync(() =>
```

Comment:
❓ What is `InvokeAsync` here? Depending on the answer, it is likely one or more of the previous comment patterns applies to this case.

Reply:
```csharp
        public static void InvokeAsync(this Control control, Action action)
        {
            InvokeAsync(control, _ => action(), null);
        }

        public static void InvokeAsync(this Control control, SendOrPostCallback action, object state)
        {
            SendOrPostCallback checkDisposedAndInvoke = (s) =>
            {
                if (!control.IsDisposed)
                    action(s);
            };

            if (!control.IsDisposed)
                UISynchronizationContext.Post(checkDisposedAndInvoke, state);
        }
```